proto/pulumi/provider.proto

```diff
@@ -751,76 +764,160 @@ message DeleteRequest {
+    // monitor.
+    //
+    // Do *not* change field IDs. Add new fields at the end with appropriate field IDs as necessary.
+
+    // An optional parent resource that the component (and by extension, its nested resources) should be children of.
+    string parent = 9;
+
+    // The component resource's input properties. Unlike the inputs of custom resources, these will *not* have been
+    // passed to a call to [](pulumirpc.ResourceProvider.Check). By virtue of their being a composition of other
+    // resources, component resources are able to (and therefore expected) to validate their own inputs. Moreover,
+    // [](pulumirpc.ResourceProvider.Check) will be called on any inputs passed to nested custom resources as usual.
+    google.protobuf.Struct inputs = 10;
+
+    // A map of property dependencies for the component resource and its nested resources.
+    map<string, PropertyDependencies> inputDependencies = 11;
+
+    // A map of package names to provider references for the component resource and its nested resources.
+    map<string, string> providers = 13;
+
+    // A list of URNs that this resource and its nested resources depend on.
+    repeated string dependencies = 15;
+
+    // A set of configuration keys whose values are [secret](output-secrets).
+    repeated string configSecretKeys = 16;
+
+    // The organization to which this resource and its nested resources will belong.
+    string organization = 17;
+
+    // True if and only if the resource (and by extension, its nested resources) should be marked as protected.
+    // Protected resources cannot be deleted without first being unprotected.
+    bool protect = 12;
+
+    // A list of additional URNs that should be considered the same as this component's URN (and which will therefore be
+    // used to build aliases for its nested resource URNs). These may be URNs that previously referred to this component
+    // e.g. if it had its parent (and consequently URN) changed.
+    repeated string aliases = 14;
+
+    // A list of input properties whose values should be treated as [secret](output-secrets).
+    repeated string additionalSecretOutputs = 18;
+
+    // A set of custom timeouts that specify how long the caller is prepared to wait for the various CRUD operations of
+    // this resource's nested resources.
+    CustomTimeouts customTimeouts = 19;
+
+    // The URN of a resource that this resource (and thus its nested resources) will be implicitly deleted with. If the
+    // resource referred to by this URN is deleted in the same operation that this resource would be deleted, the
+    // [](pulumirpc.ResourceProvider.Delete) call for this resource will be elided (since this dependency signals that
+    // it will have already been deleted).
+    string deletedWith = 20;
+
+    // If true, this resource (and its nested resources) must be deleted *before* its replacement is created.
```

Comment:
Does it really work this way? Is it possible to indicate a delete before replace plan? Does the provider decide this? I'm a bit less familiar with components TBH, maybe there are more pages that document this.